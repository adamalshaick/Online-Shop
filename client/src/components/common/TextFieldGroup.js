import React from "react";
import classnames from "classnames";
import PropTypes from "prop-types";
import TextField from "@material-ui/core/TextField";

const TextFieldGroup = ({
  name,
  placeholder,
  value,
  error,
  info,
  type,
  onChange,
  disabled
}) => {
  return (
    <div>
      <TextField
        style={{ width: "100%" }}
        label={placeholder}
        name={name}
        value={value}
        onChange={onChange}
        multiline
        rows="4"
        margin="normal"
      />
      {error ? (
        <div>
          <small style={{ color: "red" }} className="entry">
            {error}
          </small>
        </div>
      ) : null}
    </div>
  );
};

TextFieldGroup.propTypes = {
  name: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  value: PropTypes.string.isRequired,
  info: PropTypes.string,
  error: PropTypes.string,
  type: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  disabled: PropTypes.string
};

TextFieldGroup.defaultProps = {
  type: "text"
};

export default TextFieldGroup;
