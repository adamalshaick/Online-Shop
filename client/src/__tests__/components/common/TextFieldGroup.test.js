import React from "react";
import TextFieldGroup from "../../../components/common/TextFieldGroup";
import renderer from "react-test-renderer";

const wrapper = renderer.create(<TextFieldGroup />).toJSON();
describe("render component", () => {
  it("renders component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
