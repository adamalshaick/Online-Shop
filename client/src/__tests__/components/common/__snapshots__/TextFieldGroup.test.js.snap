// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component renders component 1`] = `
<div
  className="form-group"
>
  <input
    className="form-control form-control-lg"
    type="text"
  />
</div>
`;
